cis194/week4/tien/homework4/src/Homework4.hs

```diff
@@ -0,0 +1,30 @@
+module Homework4 where
+
+-- Exercise 1 --
+-- 1.1 --
+fun1 :: [Integer] -> Integer
+fun1 [] = 1
+fun1 (x:xs)
+  | even x = (x - 2) * fun1 xs
+  | otherwise = fun1 xs
+
+fun1' :: [Integer] -> Integer
+fun1' = product . map (subtract 2) . filter (even)
```

Comment:
We normally don't use brackets when it's not necessary to reduce the characters to be read.
```suggestion
fun1' = product . map (subtract 2) . filter even
```